Error when building on Ubuntu 18.04

Exokit v0.0.517
Ubuntu 18.04
Node v11.6.0 ( https://github.com/webmixedreality/exokit/blob/d2568b4b7228586edcba1d31b98a8095f19ff2a6/Dockerfile#L13 )
gcc 7.3.0
g++ 7.3.0

commands ran:
```
git clone git@github.com:webmixedreality/exokit.git
npm install
```

---------------------------------------------------------
 
highlighted errors in full log - https://gist.github.com/chrislatorres/cb9b6b33504c2c528bb739f58091b0de#file-gistfile1-txt-L6595-L6729 
  
npm log - https://gist.github.com/chrislatorres/2f79ac06bf4a24fe0b1cd8a2c9c28232 
  

---------------------------------------------------------

  
```
../deps/openvr/src/ivrcompositor.cpp:20:19: error: ‘function’ is not a member of ‘std’
   std::deque<std::function<void()>> reqCbs;
                   ^~~~~~~~
../deps/openvr/src/ivrcompositor.cpp:20:19: note: suggested alternative: ‘is_function’
   std::deque<std::function<void()>> reqCbs;
                   ^~~~~~~~
                   is_function
../deps/openvr/src/ivrcompositor.cpp:20:19: error: ‘function’ is not a member of ‘std’
../deps/openvr/src/ivrcompositor.cpp:20:19: note: suggested alternative: ‘is_function’
   std::deque<std::function<void()>> reqCbs;
                   ^~~~~~~~
                   is_function
../deps/openvr/src/ivrcompositor.cpp:20:33: error: template argument 1 is invalid
   std::deque<std::function<void()>> reqCbs;
                                 ^
../deps/openvr/src/ivrcompositor.cpp:20:33: error: template argument 2 is invalid
../deps/openvr/src/ivrcompositor.cpp:20:34: error: expected unqualified-id before ‘>’ token
   std::deque<std::function<void()>> reqCbs;
                                  ^~
../deps/openvr/src/ivrcompositor.cpp:21:19: error: ‘function’ is not a member of ‘std’
   std::deque<std::function<void()>> resCbs;
                   ^~~~~~~~
../deps/openvr/src/ivrcompositor.cpp:21:19: note: suggested alternative: ‘is_function’
   std::deque<std::function<void()>> resCbs;
                   ^~~~~~~~
                   is_function
../deps/openvr/src/ivrcompositor.cpp:21:19: error: ‘function’ is not a member of ‘std’
../deps/openvr/src/ivrcompositor.cpp:21:19: note: suggested alternative: ‘is_function’
   std::deque<std::function<void()>> resCbs;
                   ^~~~~~~~
                   is_function
../deps/openvr/src/ivrcompositor.cpp:21:33: error: template argument 1 is invalid
   std::deque<std::function<void()>> resCbs;
                                 ^
../deps/openvr/src/ivrcompositor.cpp:21:33: error: template argument 2 is invalid
../deps/openvr/src/ivrcompositor.cpp:21:34: error: expected unqualified-id before ‘>’ token
   std::deque<std::function<void()>> resCbs;
                                  ^~
../deps/openvr/src/ivrcompositor.cpp: In function ‘void vr::RunResInMainThread(uv_async_t*)’:
../deps/openvr/src/ivrcompositor.cpp:27:10: error: ‘function’ is not a member of ‘std’
     std::function<void()> resCb;
          ^~~~~~~~
../deps/openvr/src/ivrcompositor.cpp:27:10: note: suggested alternative: ‘is_function’
     std::function<void()> resCb;
          ^~~~~~~~
          is_function
../deps/openvr/src/ivrcompositor.cpp:27:27: error: ‘resCb’ was not declared in this scope
     std::function<void()> resCb;
                           ^~~~~
../deps/openvr/src/ivrcompositor.cpp:31:15: error: ‘resCbs’ was not declared in this scope
       resCb = resCbs.front();
               ^~~~~~
../deps/openvr/src/ivrcompositor.cpp: In lambda function:
../deps/openvr/src/ivrcompositor.cpp:70:12: error: ‘function’ is not a member of ‘std’
       std::function<void()> reqCb;
            ^~~~~~~~
../deps/openvr/src/ivrcompositor.cpp:70:12: note: suggested alternative: ‘is_function’
       std::function<void()> reqCb;
            ^~~~~~~~
            is_function
../deps/openvr/src/ivrcompositor.cpp:70:29: error: ‘reqCb’ was not declared in this scope
       std::function<void()> reqCb;
                             ^~~~~
../deps/openvr/src/ivrcompositor.cpp:74:17: error: ‘reqCbs’ is not a member of ‘vr’
         if (vr::reqCbs.size() > 0) {
                 ^~~~~~
../deps/openvr/src/ivrcompositor.cpp:74:17: note: suggested alternative: ‘reqSem’
         if (vr::reqCbs.size() > 0) {
                 ^~~~~~
                 reqSem
../deps/openvr/src/ivrcompositor.cpp:75:23: error: ‘reqCbs’ is not a member of ‘vr’
           reqCb = vr::reqCbs.front();
                       ^~~~~~
../deps/openvr/src/ivrcompositor.cpp:75:23: note: suggested alternative: ‘reqSem’
           reqCb = vr::reqCbs.front();
                       ^~~~~~
                       reqSem
../deps/openvr/src/ivrcompositor.cpp:76:15: error: ‘reqCbs’ is not a member of ‘vr’
           vr::reqCbs.pop_front();
               ^~~~~~
../deps/openvr/src/ivrcompositor.cpp:76:15: note: suggested alternative: ‘reqSem’
           vr::reqCbs.pop_front();
               ^~~~~~
               reqSem
../deps/openvr/src/ivrcompositor.cpp: In static member function ‘static Nan::NAN_METHOD_RETURN_TYPE IVRCompositor::WaitGetPoses(Nan::NAN_METHOD_ARGS_TYPE)’:
../deps/openvr/src/ivrcompositor.cpp:147:57: warning: overflow in implicit constant conversion [-Woverflow]
   memset(hmdArray, std::numeric_limits<float>::quiet_NaN(), 16);
                    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~
../deps/openvr/src/ivrcompositor.cpp:148:68: warning: overflow in implicit constant conversion [-Woverflow]
  memset(leftControllerArray, std::numeric_limits<float>::quiet_NaN(), 16);
                              ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~
../deps/openvr/src/ivrcompositor.cpp:149:69: warning: overflow in implicit constant conversion [-Woverflow]
 memset(rightControllerArray, std::numeric_limits<float>::quiet_NaN(), 16);
                              ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~
../deps/openvr/src/ivrcompositor.cpp: In static member function ‘static Nan::NAN_METHOD_RETURN_TYPE IVRCompositor::RequestGetPoses(Nan::NAN_METHOD_ARGS_TYPE)’:
../deps/openvr/src/ivrcompositor.cpp:221:9: error: ‘reqCbs’ is not a member of ‘vr’
     vr::reqCbs.push_back([obj, system, hmdArray, leftControllerArray, rightControllerArray, vrPoseRes]() -> void {
         ^~~~~~
../deps/openvr/src/ivrcompositor.cpp:221:9: note: suggested alternative: ‘reqSem’
     vr::reqCbs.push_back([obj, system, hmdArray, leftControllerArray, rightControllerArray, vrPoseRes]() -> void {
         ^~~~~~
         reqSem
../deps/openvr/src/ivrcompositor.cpp: In lambda function:
../deps/openvr/src/ivrcompositor.cpp:225:61: warning: overflow in implicit constant conversion [-Woverflow]
       memset(hmdArray, std::numeric_limits<float>::quiet_NaN(), 16);
                        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~
../deps/openvr/src/ivrcompositor.cpp:226:72: warning: overflow in implicit constant conversion [-Woverflow]
  memset(leftControllerArray, std::numeric_limits<float>::quiet_NaN(), 16);
                              ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~
../deps/openvr/src/ivrcompositor.cpp:227:73: warning: overflow in implicit constant conversion [-Woverflow]
 memset(rightControllerArray, std::numeric_limits<float>::quiet_NaN(), 16);
                              ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~
../deps/openvr/src/ivrcompositor.cpp:279:13: error: ‘resCbs’ is not a member of ‘vr’
         vr::resCbs.push_back([vrPoseRes]() -> void {
             ^~~~~~
../deps/openvr/src/ivrcompositor.cpp: In static member function ‘static Nan::NAN_METHOD_RETURN_TYPE IVRCompositor::Submit(Nan::NAN_METHOD_ARGS_TYPE)’:
../deps/openvr/src/ivrcompositor.cpp:316:41: warning: ‘uint32_t v8::Value::Uint32Value() const’ is deprecated: Use maybe version [-Wdeprecated-declarations]
   GLuint texture = info[1]->Uint32Value();
                                         ^
In file included from /home/chris/.node-gyp/10.15.0/include/node/v8.h:26:0,
                 from /home/chris/.node-gyp/10.15.0/include/node/node.h:63,
                 from /home/chris/github/exokit/node_modules/nan/nan.h:53,
                 from /home/chris/github/exokit/deps/openvr/include/ivrcompositor.h:4,
                 from ../deps/openvr/src/ivrcompositor.cpp:1:
/home/chris/.node-gyp/10.15.0/include/node/v8.h:2477:47: note: declared here
   V8_DEPRECATED("Use maybe version", uint32_t Uint32Value() const);
                                               ^
/home/chris/.node-gyp/10.15.0/include/node/v8config.h:324:3: note: in definition of macro ‘V8_DEPRECATED’
   declarator __attribute__((deprecated(message)))
   ^~~~~~~~~~
exokit.target.mk:234: recipe for target 'Release/obj.target/exokit/deps/openvr/src/ivrcompositor.o' failed
make: *** [Release/obj.target/exokit/deps/openvr/src/ivrcompositor.o] Error 1
make: Leaving directory '/home/chris/github/exokit/build'
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/home/chris/.nvm/versions/node/v10.15.0/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:262:23)
gyp ERR! stack     at ChildProcess.emit (events.js:182:13)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:240:12)
gyp ERR! System Linux 4.15.0-46-generic
gyp ERR! command "/home/chris/.nvm/versions/node/v10.15.0/bin/node" "/home/chris/.nvm/versions/node/v10.15.0/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /home/chris/github/exokit
gyp ERR! node -v v10.15.0
gyp ERR! node-gyp -v v3.8.0
gyp ERR! not ok 
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! exokit@0.0.517 install: `node ./scripts/preinstall.js && node-gyp rebuild`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the exokit@0.0.517 install script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/chris/.npm/_logs/2019-03-05T21_00_02_541Z-debug.log
```

--- a/deps/openvr/src/ivrcompositor.cpp
+++ b/deps/openvr/src/ivrcompositor.cpp
@@ -7,6 +7,7 @@
 #include <node.h>
 #include <openvr.h>
 #include <ivrsystem.h>
+#include <functional>
 
 using namespace v8;
 

